source/dub/generators/targetdescription.d

```diff
@@ -47,6 +47,7 @@ class TargetDescriptionGenerator : ProjectGenerator {
 			d.buildSettings = ti.buildSettings.dup;
 			d.dependencies = ti.dependencies.dup;
 			d.linkDependencies = ti.linkDependencies.dup;
+			if (settings.lowmem) d.buildSettings.options |= BuildOption.lowmem;
```

Comment:
It looks like it will, but it actually isn't.
I looked into the reasons why it isn't.

In L47, the `target` is used.
https://github.com/shoo/dub/blob/a0090af56f062d67fc949aa6beee01198317d4b0/source/dub/generators/targetdescription.d#L47

The settings of `target` look as if they were passed from `GeneratorSettings.buildSettings` on this line:
https://github.com/shoo/dub/blob/a0090af56f062d67fc949aa6beee01198317d4b0/source/dub/generators/generator.d#L107

If I look inside the function, the following line shows the addition of settings:
https://github.com/shoo/dub/blob/a0090af56f062d67fc949aa6beee01198317d4b0/source/dub/generators/generator.d#L560

However, there is no code in this function to add `options` (and `requirements`).
https://github.com/shoo/dub/blob/a0090af56f062d67fc949aa6beee01198317d4b0/source/dub/compilers/buildsettings.d#L67-L90

If I add the code to add the options here, I get the expected behavior.

```d
		...
		addPreRunCommands(bs.preRunCommands);
		addPostRunCommands(bs.postRunCommands);
		addRequirements(bs.requirements);
		addOptions(bs.options);
```

However, I can't speculate on the extent of the impact of making this change.
Is this change reasonable?

Reply:
hm no I don't think it's a good place in the add function. I think it should work if it's added into https://github.com/dlang/dub/blob/800dd8031ef1a640ca95ef874a5cb3b653f4c1fd/source/dub/project.d#L1148

Reply:
The role of `processVar` seems to be to expand the variables contained in dub.sdl/dub.json, which exist in each project by using GeneratorSettings, etc.
The purpose of this PR, to transfer the lowmem option included in GeneratorSettings to BuildSettings, seems to be contrary to the purpose of this function.

Another idea is calling `BuildSettings.addOptions` immediately after the `processVar` execution. What do you think of this idea?

Reply:
hm that's true but the processVars function already adds options from other, which doesn't have anything to do with that, so I think it would be fine.

cc @s-ludwig